src/core/server/saved_objects/migrationsv2/integration_tests/actions.test.ts

```diff
@@ -904,8 +899,87 @@ describe('migration actions', () => {
       `);
     });
 
+    it('only returns documents that match the provided "query"', async () => {
+      const openPitTask = openPit(client, 'existing_index_with_docs');
+      const pitResponse = (await openPitTask()) as Either.Right<OpenPitResponse>;
+
+      const readWithPitTask = readWithPit(
+        client,
+        pitResponse.right.pitId,
+        {
+          match: { title: { query: 'doc' } },
+        },
+        1000,
+        undefined
+      );
+
+      const docsResponse = (await readWithPitTask()) as Either.Right<ReadWithPit>;
+
+      expect(docsResponse.right.outdatedDocuments.map((doc) => doc._source.title).sort())
+        .toMatchInlineSnapshot(`
+        Array [
+          "doc 1",
+          "doc 2",
+          "doc 3",
+        ]
+      `);
+    });
+
+    it('returns docs with _seq_no and _primary_term when specified', async () => {
+      const openPitTask = openPit(client, 'existing_index_with_docs');
+      const pitResponse = (await openPitTask()) as Either.Right<OpenPitResponse>;
+
+      const readWithPitTask = readWithPit(
+        client,
+        pitResponse.right.pitId,
+        {
+          match: { title: { query: 'doc' } },
+        },
+        1000,
+        undefined,
+        true
+      );
+
+      const docsResponse = (await readWithPitTask()) as Either.Right<ReadWithPit>;
+
+      expect(docsResponse.right.outdatedDocuments).toEqual(
+        expect.arrayContaining([
+          expect.objectContaining({
+            _seq_no: expect.any(Number),
+            _primary_term: expect.any(Number),
+          }),
+        ])
+      );
+    });
+
+    it('doesn not return docs with _seq_no and _primary_term if not specified', async () => {
+      const openPitTask = openPit(client, 'existing_index_with_docs');
+      const pitResponse = (await openPitTask()) as Either.Right<OpenPitResponse>;
+
+      const readWithPitTask = readWithPit(
+        client,
+        pitResponse.right.pitId,
+        {
+          match: { title: { query: 'doc' } },
+        },
+        1000,
+        undefined
+      );
+
+      const docsResponse = (await readWithPitTask()) as Either.Right<ReadWithPit>;
+
+      expect(docsResponse.right.outdatedDocuments).toEqual(
+        expect.arrayContaining([
+          expect.not.objectContaining({
+            _seq_no: expect.any(Number),
+            _primary_term: expect.any(Number),
+          }),
+        ])
```

Comment:
NIT: Test will pass if `_seq_no` or `_primary_term` exist but it not a number. I would rather test for property existence here. OTOH, by the ES spec, these properties are numbers, so it's probably fine as it is.